flutter/lib/data/export_result.dart

```diff
@@ -0,0 +1,123 @@
+
+  final String id;
+  final String throughput;
+  final String accuracy;
+  // numeric values are saved as string to match old android app behavior
+  final String minDuration;
+  final String duration;
+  final String minSamples;
+  final String numSamples;
+  final int shardsNum;
+  final int batchSize;
+  final String mode;
+  final String datetime;
+  final String backendName;
+
+  ExportResult(
+      {required this.id,
+      required this.throughput,
+      required this.accuracy,
+      required this.minDuration,
+      required this.duration,
+      required this.minSamples,
+      required this.numSamples,
+      required this.shardsNum,
+      required this.batchSize,
+      required this.mode,
+      required this.datetime,
+      required this.backendName});
+
+  ExportResult.fromRunInfo(RunInfo info)
+      : this(
+            id: info.settings.benchmark_id,
+            throughput: info.runMode == BenchmarkRunMode.accuracy
+                ? 'N/A'
+                : info.result.throughput.toString(),
+            accuracy: info.runMode == BenchmarkRunMode.accuracy
+                ? info.result.accuracy
+                : 'N/A',
+            minDuration: info.settings.min_duration.toString(),
+            duration: info.result.durationMs.toString(),
+            minSamples: info.settings.min_query_count.toString(),
+            numSamples: info.result.numSamples.toString(),
+            shardsNum: info.settings.threads_number,
+            batchSize: info.settings.batch_size,
+            mode: info.runMode.getResultModeString(),
+            datetime: DateTime.now().toIso8601String(),
+            backendName: info.result.backendName);
+
+  ExportResult.fromJson(Map<String, dynamic> json)
+      : this(
+            id: json[_tagId] as String,
+            throughput: json[_tagThroughput] as String,
+            accuracy: json[_tagAccuracy] as String,
+            minDuration: json[_tagMinDuration] as String,
+            duration: json[_tagDuration] as String,
+            minSamples: json[_tagMinSamples] as String,
+            numSamples: json[_tagNumSamples] as String,
+            shardsNum: json[_tagSharsNum] as int,
+            batchSize: json[_tagBatchSize] as int,
+            mode: json[_tagMode] as String,
+            datetime: json[_tagDatetime] as String,
+            backendName: json[_tagBackendName] as String);
+
+  Map<String, dynamic> toJson() => {
+        _tagId: id,
+        _tagConfiguration: {
+          _tagRuntime: '',
+        },
+        _tagThroughput: throughput,
+        _tagAccuracy: accuracy,
+        _tagMinDuration: minDuration,
+        _tagDuration: duration,
+        _tagMinSamples: minSamples,
+        _tagNumSamples: numSamples,
+        _tagSharsNum: shardsNum,
+        _tagBatchSize: batchSize,
+        _tagMode: mode,
+        _tagDatetime: datetime,
+        _tagBackendName: backendName,
+      };
+}
+
+class ExportResultList {
```

Comment:
I think it quite common to use singular noun for one object and plural noun for a list of objects, so `ExportResults` should be sufficient.

Reply:
I wasn't able to find too much info on best practices of naming collections.
One article I found is here: https://medium.com/swlh/an-opinionated-guide-to-naming-your-code-aimed-at-new-developers-6f769d60e098
It looks very reasonable to me.
It suggests always using singular nouns for class names. Plural nouns are fine for variable names but in class names they are harder to understand.

Reply:
Naming convention is more or less subjective, so I am fine with any style if it's consistent. We already have some class with plural noun:
- Icons
- AppColors
- RunSettings
- ResultKeys
- ShareKeys
- ..

Reply:
The difference between `ExportResults` and other plural class names in the project is the existence of `ExportResult` class.
We don't have a `ShareKey`, `RunSetting`, or `AppColor` class. If we had any of those, I would argue that having both singular and plural class name is confusing.
Also, `Icons` is a bad name in my opinion, both because it collides with the standard library name and because there is an `Icon` class.

Reply:
Yes. We need some more refactoring to make the naming consistent.